pkg/blockchain/ethAppChainReader.go

```diff
@@ -0,0 +1,139 @@
+package blockchain
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/ethclient"
+	gm "github.com/xmtp/xmtpd/pkg/abi/groupmessagebroadcaster"
+	iu "github.com/xmtp/xmtpd/pkg/abi/identityupdatebroadcaster"
+	"github.com/xmtp/xmtpd/pkg/utils"
+)
+
+func buildMessagesTopic() (common.Hash, error) {
+	abi, err := gm.GroupMessageBroadcasterMetaData.GetAbi()
+	if err != nil {
+		return common.Hash{}, err
+	}
+	return utils.GetEventTopic(abi, "MessageSent")
+}
+
+func buildIdentityUpdatesTopic() (common.Hash, error) {
+	abi, err := iu.IdentityUpdateBroadcasterMetaData.GetAbi()
+	if err != nil {
+		return common.Hash{}, err
+	}
+	return utils.GetEventTopic(abi, "IdentityUpdateCreated")
+}
+
+// Implements AppChainReader
```

Comment:
The interface implementation can be forced with `var _ AppChainReader = &EthAppChainReader{}`. That way the code won't compile if the type doesn't implement the interface.